src/event/LogEvent.php

```diff
@@ -109,6 +115,22 @@ private function verify() {
         $this->ensureSet($this->timestamp, 'Timestamp');
     }
 
+    private static function buildCidFromIPAndUserAgent($ipAddress, $userAgent) {
+        return hash('md5', $ipAddress . $userAgent);
+    }
+
+    private static function buildIpGroupHashes($ipAddress) {
+        $ipComponents = explode('.', $ipAddress);
```

Comment:
You just want to hash the components of IPv4-Addresses?

Reply:
We need both, hash over the complete IP address and hashes over the IP groups. This is to make the bot detection algorithm in the backend more efficient.

Reply:
Sorry I missed to mention that I meant IPv6. This method would not handle those.